lib/msgpax.ex

```diff
@@ -27,6 +27,27 @@ defmodule Msgpax do
 
   """
 
+  @type pack_error_reason ::
+    {:too_big, any} |
+    {:not_encodable, any}
+
+  @type unpack_error_reason ::
+    {:extra_bytes, binary} |
+    {:bad_format, binary} |
+    :incomplete |
+    {:not_supported_ext, integer} |
+    {:ext_unpacker_failed, Msgpax.Ext.type, module, binary}
+
+  @type unpacked_value ::
```

Comment:
I think we don't need this type as unpacked value can be anything if `ext` option is provided.